### PR TITLE
Adding fakes test package to fake out RBE server.

### PR DESCRIPTION
**Pull request by ola-rozenfeld**
If you prefer, I can break it down further and only introduce the fake CAS here, while introducing the fake ActionCache and Exec in a separate CL.

### CODE REVIEW

**Comment by ola-rozenfeld** [comment]
Commit with everything: 50b27cf#diff-ecad5dc6d2d82f0cc91e81469ff16b37

File: go/internal/test/fakes/cas.go [hunk 1]
```diff
@@ -172,33 +176,86 @@ func (f *fakeWriter) Write(stream bsgrpc.ByteStream_WriteServer) (err error) {
...
+	reads     map[digest.Digest]int
+	writes    map[digest.Digest]int
 	mu        sync.RWMutex
 	batchReqs int
 	writeReqs int
 }
 
-func (f *fakeCAS) FindMissingBlobs(ctx context.Context, req *repb.FindMissingBlobsRequest) (*repb.FindMissingBlobsResponse, error) {
+// NewCas returns a new empty fake CAS.
+func NewCas() *CAS {
```
[the remark this answers is not in the document]
Comment (ola-rozenfeld) [reply go/internal/test/fakes/cas.go]: Done.

File: go/internal/test/fakes/ac.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+	"github.com/bazelbuild/remote-apis-sdks/go/digest"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	repb "github.com/bazelbuild/remote-apis/build/bazel/remote/execution/v2"
+)
+
+// ActionCache implements the RE ActionCache interface, storing fixed results.
+type ActionCache struct {
```
[the remark this answers is not in the document]
Comment (ola-rozenfeld) [reply go/internal/test/fakes/ac.go]: Can't hurt! Done.

File: go/internal/test/fakes/ac.go [hunk 3]
```diff
@@ -66,6 +74,8 @@ func (c *ActionCache) GetWrites(d digest.Digest) int {
 
 // GetActionResult returns a stored result, if it was found.
 func (c *ActionCache) GetActionResult(ctx context.Context, req *repb.GetActionResultRequest) (res *repb.ActionResult, err error) {
+	c.mu.RLock()
```
Comment (ramymedhat) [inline go/internal/test/fakes/ac.go]: This updates the reads map so should be a standard Lock

File: go/internal/test/fakes/cas.go [hunk 4]
```diff
@@ -223,6 +223,8 @@ func (f *CAS) Clear() {
 
 // Put adds a given blob to the cache and returns its digest.
 func (f *CAS) Put(blob []byte) digest.Digest {
+	f.mu.RLock()
```
Comment (ramymedhat) [inline go/internal/test/fakes/cas.go]: Lock()?

File: go/internal/test/fakes/cas.go [hunk 5]
```diff
@@ -402,6 +404,8 @@ func (f *CAS) Write(stream bsgrpc.ByteStream_WriteServer) (err error) {
 
 // Read implements the corresponding RE API function.
 func (f *CAS) Read(req *bspb.ReadRequest, stream bsgrpc.ByteStream_ReadServer) error {
+	f.mu.RLock()
```
Comment (ramymedhat) [inline go/internal/test/fakes/cas.go]: Same here?

File: go/internal/test/fakes/ac.go (hunk 3 above)
Comment (ola-rozenfeld) [reply go/internal/test/fakes/ac.go]: Done.

File: go/internal/test/fakes/cas.go (hunk 4 above)
Comment (ola-rozenfeld) [reply go/internal/test/fakes/cas.go]: Done.

File: go/internal/test/fakes/cas.go (hunk 5 above)
Comment (ola-rozenfeld) [reply go/internal/test/fakes/cas.go]: Done.

**Pull request merged**